pkg/api/v1/control-plane/server/generic_server.go

```diff
@@ -21,14 +21,14 @@ import (
 	"strconv"
 	"sync/atomic"
 
-	envoy_api_v2 "github.com/envoyproxy/go-control-plane/envoy/api/v2"
 	envoy_config_core_v3 "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
-	envoy_service_discovery_v2 "github.com/envoyproxy/go-control-plane/envoy/service/discovery/v2"
 	envoy_service_discovery_v3 "github.com/envoyproxy/go-control-plane/envoy/service/discovery/v3"
 	"github.com/golang/protobuf/proto"
 	"github.com/golang/protobuf/ptypes/any"
+	envoy_api_v2 "github.com/solo-io/solo-kit/pkg/api/external/envoy/api/v2"
 	"github.com/solo-io/solo-kit/pkg/api/v1/control-plane/resource"
 	"github.com/solo-io/solo-kit/pkg/api/v1/control-plane/util"
+	gloo_discovery "github.com/solo-io/solo-kit/pkg/api/xds"
```

Comment:
This comment has less to do with this import and more to do with the generic server in general. 
What do you think about introducing a new StreamGloo interface that sends and receives gloo discovery requests and gloo discovery responses? We can then remove the StreamV2 interface and there's a clear distinction between the gloo stream (that will not change) and the envoy stream (that will change when we upgrade).